### ISSUE TITLE
Encoding and collation for just one column

### ISSUE THREAD

**Issue by TrogWarZ** [cakephp/phinx#661]
Hello.

I'm trying to set `utf8_general_ci` for whole table but `utf8mb4_unicode_ci` for just one column like this:

``` php
$this->table('foo')
    ->setOptions([
        'encoding'  => 'utf8',
        'collation' => 'utf8_general_ci',
    ])
    ->addColumn('text_one', 'string')
    ->addColumn('text_two', 'string', ['collation' => 'utf8mb4_unicode_ci''])
    ->save()
;
```

And get `[RuntimeException] "collation" is not a valid column option.`

But why? I can use SQL (tested for MySQL) like this and it will works as expected:

``` sql
CREATE TABLE `foo` (
    `id` int(11) NOT NULL AUTO_INCREMENT,
    `text_one` varchar(255) COLLATE utf8_general_ci NOT NULL,
    `text_two` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
    PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci;
```

Is this bug in Phinx? Or unimplemented feature and i must use raw-sql queries after creating table?

Searching in docs by "collation" keyword shows only page for configuration phinx.yml.

**Comment by TrogWarZ** [comment]
Just see sources and add this unsupported feature.

**Comment by J-Fricke** [comment]
Is this still a feature waiting to be added? Need this ability.

**Comment by javiergloria** [comment]
hey @J-Fricke  how did you manage to add the collations for just one column ?

**Comment by rquadling** [comment]
Hi. If anyone wants to pick up the PR#662 and rebase it on the latest dev, then I can take a look at it

**Comment by TrogWarZ** [comment]
Moved code from earlier PR to new one.

**Comment by J-Fricke** [comment]
@javiergloria 
I ended up just using some SQL like this after setting it for the whole table like the OP has above:
`$this->execute('ALTER TABLE address MODIFY company VARCHAR (75) CHARACTER SET utf8;');`

**Comment by pburkind** [comment]
$this->table->setOptions is not discussed in the documentation

**Comment by inoas** [comment]
This works for changeColumn() for sure... can you confirm that it got fixed for addColumn too?
Example

```php
	public function change() {
		$table = $this->table('queue_processes');
		$table->changeColumn('pid', 'text', [
			'encoding' => 'ascii',
			'collation' => 'ascii_bin',
		]);

		$table = $this->table('queued_jobs');
		$table->changeColumn('data', 'text', [
			'encoding' => 'utf8mb4',
			'collation' => 'utf8mb4_bin',
		]);
		$table->changeColumn('job_group', 'text', [
			'encoding' => 'utf8mb4',
			'collation' => 'utf8mb4_bin',
		]);
		$table->changeColumn('reference', 'text', [
			'encoding' => 'utf8mb4',
			'collation' => 'utf8mb4_bin',
		]);
		$table->changeColumn('failure_message', 'text', [
			'encoding' => 'utf8mb4',
			'collation' => 'utf8mb4_bin',
		]);
		$table->changeColumn('workerkey', 'text', [
			'encoding' => 'utf8mb4',
			'collation' => 'utf8mb4_bin',
		]);
		$table->changeColumn('status', 'text', [
			'encoding' => 'utf8mb4',
			'collation' => 'utf8mb4_bin',
		]);
	}
```

**Comment by dereuromark** [comment]
Are you able to provide a patch with your suggested changes as PR?

**Comment by MasterOdin** [comment]
@dereuromark this can be closed as it was implemented in 0.7.0 (in commit 39e99f9db0ff6303d8079a70baf7f9e6a15d64f9) and is documented.
